field_audit/migrations/0001_initial.py

```diff
@@ -0,0 +1,45 @@
+from django.contrib.postgres.indexes import GinIndex
+from django.db import migrations, models
+
+from ..models import get_date
+
+
+class Migration(migrations.Migration):
+
+    initial = True
+
+    dependencies = []
+
+    operations = [
+        migrations.CreateModel(
+            name='AuditEvent',
+            fields=[
+                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),  # noqa: E501
+                ('event_date', models.DateTimeField(db_index=True, default=get_date)),  # noqa: E501
+                ('object_class_path', models.CharField(db_index=True, max_length=255)),  # noqa: E501
+                ('object_pk', models.JSONField()),
+                ('changed_by', models.JSONField()),
+                ('is_create', models.BooleanField(default=False)),
+                ('is_delete', models.BooleanField(default=False)),
+                ('delta', models.JSONField()),
+            ],
+        ),
+        migrations.AddIndex(
+            model_name='auditevent',
+            index=GinIndex(
```

Comment:
What kind of queries is this index intended to support? Wondering if it needs to be a full GIN index or if it could using `json_path_ops` or even be a B-Tree index if we only need to know if a specific key is present in the data.

Reply:
Ah, I added this and completely forgot about it (I meant to remove it). I did some testing with indexes into `jsonb` columns. We don't want an index on the `delta` column. There is no known audit query "practice" that requires it: that is, the "expected" queries on this table will be something like:

```python
AuditEvent.objects.filter(object_class_path='app.models.Thing', changed_by__username='someone')
```

When considering HQ specifically (which adds an `HQAuditor` for capturing `changed_by["domain"]`, the "happy path" queries are expected to look something like:

```python
AuditEvent.objects.filter(object_class_path='app.models.Thing', changed_by__domain='the-things')
```

This means that HQ will want to add its own index (perhaps a `GIN`?). For example:

```sql
CREATE INDEX field_audit_auditevent_changed_by_domain
  ON field_audit_auditevent
  USING GIN (changed_by->'domain' jsonb_path_ops);
```
Still thinking on this, a `HASH` index might be better because `changed_by->'username'` (and other `changed_by` keys like `domain`) are not nested objects (the values are just strings), so a `HASH` index might be better suited here. I'll be testing this before I add anything. Feedback is very welcome.

Reply:
Index removed in c5d1300.